extensions/azdata/src/test/common/httpClient.test.ts

```diff
@@ -29,14 +29,14 @@ describe('HttpClient', function () {
 	});
 
 	it('downloads file successfully', async function (): Promise<void> {
-		const downloadPath = path.join(os.tmpdir(), `azdata-httpClientTest-${uuid.v4()}.txt`);
-		await HttpClient.download('https://raw.githubusercontent.com/microsoft/azuredatastudio/main/README.md', downloadPath, outputChannelMock.object);
+		const downloadFolder = path.join(os.tmpdir());
+		const downloadPath = await HttpClient.download('https://raw.githubusercontent.com/microsoft/azuredatastudio/main/README.md', downloadFolder, outputChannelMock.object);
```

Comment:
Can a download of a valid file be mocked with nock instead of having an external dependency? This server may be available during a test run leading to undesirable failure.

Reply:
Yup!